plugins/modules/foreman_realm.py

```diff
@@ -74,92 +57,36 @@
     password: "changeme"
     server_url: "https://foreman.example.com"
     name: "EXAMPLE.COM"
-    realm_proxy: 1
+    realm_proxy: "foreman.example.com"
     realm_type: "Red Hat Identity Management"
     state: present
 '''
 
-RETURN = '''# '''
-
-try:
-    from ansible.module_utils.ansible_nailgun_cement import (
-        create_server,
-        ping_server,
-        entity_mixins,
-        find_entities,
-        naildown_entity_state,
-    )
-    from nailgun.entities import Realm
-
-    has_import_error = False
-except ImportError as e:
-    has_import_error = True
-    import_error_msg = str(e)
-
-
-from ansible.module_utils.basic import AnsibleModule
+RETURN = ''' # '''
 
-
-def sanitize_realm_dict(realm_dict):
-    # This is the only true source for names (and conversions thereof)
-    name_map = {
-        'name': 'name',
-        'realm_proxy': 'realm_proxy',
-        'realm_type': 'realm_type',
-    }
-    result = {}
-    for key, value in name_map.items():
-        if key in realm_dict:
-            result[value] = realm_dict[key]
-    return result
+from ansible.module_utils.foreman_helper import ForemanEntityApypieAnsibleModule
 
 
 def main():
-
-    module = AnsibleModule(
-        argument_spec=dict(
-            server_url=dict(required=True),
-            username=dict(required=True, no_log=True),
-            password=dict(required=True, no_log=True),
-            verify_ssl=dict(type='bool', default=True),
+    module = ForemanEntityApypieAnsibleModule(
+        entity_spec=dict(
             name=dict(required=True),
-            realm_proxy=dict(type='int', required=True),
+            realm_proxy=dict(type='entity', flat_name='realm_proxy_id', required=True),
```

Comment:
This needs docs updates, as you changed from ID to name. Also backward incompatible change, but I think that's fine.